streams/src/main/java/org/apache/kafka/streams/processor/internals/StateManagerUtil.java

```diff
@@ -58,6 +58,9 @@ static boolean checkpointNeeded(final boolean enforceCheckpoint,
             return false;
         }
 
+        if (enforceCheckpoint)
```

Comment:
Could we move this logic to the `StandbyTask` only? It is the only case I have seen which could have null snapshot passed in, which could make this helper assume both snapshots are not null.
```
if (oldOffsetSnapshot == null) {
            return false;	          
}	        
```

Reply:
The null snapshot is for both active / standby tasks before the initializeIfNeeded is triggered, in that case, we should not overwrite the checkpoint even if it was enforced.